packages/documentsearch/src/interfaces.ts

```diff
@@ -94,18 +94,20 @@ export interface ISearchMatch {
 }
 
 /**
- * This interface is meant to enforce that SearchProviders implement the static
- * canSearchOn function.
+ * This interface is meant to enforce that SearchProviders implement
+ * the static canSearchOn function.
  */
-export interface ISearchProviderConstructor {
-  new (): ISearchProvider;
+export interface ISearchProviderConstructor<T extends Widget = Widget> {
+  new (): ISearchProvider<T>;
   /**
-   * Report whether or not this provider has the ability to search on the given object
+   * Report whether or not this provider has the ability to search on the
+   * given object. The function is a type guard, meaning that it returns
+   * a boolean, but has a type predicate (`x is T`) for its return signature.
```

Comment:
I'm not sure if this makes sense to add to the docstring. Mainly because
- it will be confusing to any who consume it from pure JS
- it explains a TS language feature

Reply:
It's a docstring on an interface, so it can't be consumed from pure JS. I added the note about it to help future implementers of search providers, since user-defined type guards seem to be a somewhat obscure TS feature.

If you think it's unnecessary I'll get rid of it.

Reply:
> It's a docstring on an interface

Good point, let's keep it then :)